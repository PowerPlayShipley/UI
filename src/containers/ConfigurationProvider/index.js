import React, { useState } from "react";
import PropTypes from "prop-types";

import ConfigurationContext from "../../contexts/ConfigurationContext";

const ConfigurationProvider = ({ children, configuration: initial }) => {
  // TODO: Set this up when the API is working
  const [configuration, _] = useState(initial)

  return (
    <ConfigurationContext.Provider value={configuration}>
      {children}
    </ConfigurationContext.Provider>
  )
}

ConfigurationProvider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element)
  ]),
  configuration: PropTypes.shape({
    players: PropTypes.object,
    fines: PropTypes.object,
    meta: PropTypes.object,
  }).isRequired
}

export default ConfigurationProvider
