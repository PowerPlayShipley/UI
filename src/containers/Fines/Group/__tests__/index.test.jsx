/**
 * @jest-environment jsdom
 */

import React from 'react'
import { renderWithTheme, fireEvent } from '../../../../../__mocks__/theme-test-utils'

import FinesGroup from "../index";
import ConfigurationProvider from "../../../ConfigurationProvider";

const renderWithConfig = (fn, configuration, opts) => {
  return renderWithTheme((
    <ConfigurationProvider configuration={configuration}>
      {fn()}
    </ConfigurationProvider>
  ), opts)
}

const configuration = {
  players: {
    'AA': {
      name: 'Ade'
    }
  },
  fines: {
    H: {
      name: 'Hanging',
      cost: 50
    },
    R: {
      name: 'Reverse',
      cost: 50
    }
  }
}

const data = [
  {
    player: 'AA',
    games: [["H"]]
  }
]

describe('Display FinesGroup', function () {
  it('should create a valid snapshot', function () {
    const { container } = renderWithConfig(() => <FinesGroup title='Players' data={data} />, configuration)
    expect(container.firstChild).toMatchSnapshot()
  });

  it('should create valid buttons', function () {
    const handleClick = jest.fn()

    const { getByText } = renderWithConfig(() => <FinesGroup title='Players' data={data} onFineRemoved={handleClick} />, configuration)

    fireEvent.click(getByText('H').parentNode)
    expect(handleClick).toHaveBeenCalledTimes(1)
  });

  it('should create valid toolbar', function () {
    const handleClick = jest.fn()

    const { container, getByText } = renderWithConfig(() => <FinesGroup title='Players' data={data} onFineAdded={handleClick} />, configuration)

    fireEvent.click(container.querySelector('svg'))
    fireEvent.click(getByText('R').parentNode)

    expect(handleClick).toHaveBeenCalledTimes(1)
  });

  it('should create valid open modal snapshot', function () {
    const handleClick = jest.fn()

    const { container } = renderWithConfig(() => <FinesGroup title='Players' data={data} onFineAdded={handleClick} />, configuration)

    fireEvent.click(container.querySelector('svg'))
    expect(container).toMatchSnapshot()
  });
})
