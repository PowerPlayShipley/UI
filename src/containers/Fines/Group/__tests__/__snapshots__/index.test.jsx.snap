// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display FinesGroup should create a valid snapshot 1`] = `
.c2 {
  position: relative;
  display: inline-block;
  padding: 5px 16px;
  font-size: 14px;
  font-weight: 600;
  line-height: 20px;
  white-space: nowrap;
  vertical-align: middle;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid;
  border-radius: 6px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c2:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2:disabled,
.c2.disabled,
.c2[aria-disabled=true] {
  cursor: default;
}

.c2 .dropdown-caret {
  margin-left: 4px;
  opacity: 0.8;
}

.c2 .btn-icon {
  margin-right: 4px;
  color: inherit;
  vertical-align: text-bottom;
}

.c2 .btn-icon:only-child {
  margin-right: 0;
}

.c2 {
  color: #24292f;
  background-color: #f6f8fa;
  border: 1px rgba(208,215,222,0.5) solid;
  box-shadow: 0 1px 0 rgba(27,31,36,0.04),inset 0 1px 0 rgba(255,255,255,0.25);
  -webkit-transition: 0.2s cubic-bezier(0.3,0,0.5,1);
  transition: 0.2s cubic-bezier(0.3,0,0.5,1);
  -webkit-transition-property: color,background-color,border-color;
  transition-property: color,background-color,border-color;
}

.c2:hover,
.c2.hover,
[open] > .c2 {
  background-color: #eaeef2;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
}

.c2:active {
  background-color: #e1e6ec;
  -webkit-transition: none;
  transition: none;
}

.c2.selected,
.c2[aria-selected=true] {
  background-color: #e4e9ee;
  box-shadow: inset 0 1px 0 rgba(208,215,222,0.2);
}

.c2:disabled,
.c2.disabled,
.c2[aria-disabled=true] {
  color: #8c959f;
  background-color: #f6f8fa;
  cursor: default;
}

.c2:disabled .btn-icon,
.c2.disabled .btn-icon,
.c2[aria-disabled=true] .btn-icon {
  color: #8c959f;
}

.c2:focus,
.c2.focus {
  outline: none;
  box-shadow: 0 0 0 3px rgba(9,105,218,0.15);
}

.c2.primary {
  color: #ffffff;
  background-color: #1a7f37;
  border-color: rgba(208,215,222,0.5);
  box-shadow: 0 1px 0 rgba(27,31,36,0.1), inset 0 1px 0 rgba(255,255,255,0.3);
}

.c2.primary:hover,
.c2.primary.hover,
[open] > .c2.primary {
  background-color: #116329;
  border-color: rgba(208,215,222,0.5);
}

.c2.primary:active,
.c2.primary.selected,
.c2.primary[aria-selected=true] {
  background-color: #105a25;
  box-shadow: inset 0 1px 0 rgba(0,45,17,0.2);
}

.c2.primary:disabled,
.c2.primary.disabled,
.c2.primary[aria-disabled=true] {
  color: rgba(255,255,255,0.8);
  background-color: #94d3a2;
  border-color: rgba(208,215,222,0.5);
}

.c2.primary:disabled .btn-icon,
.c2.primary.disabled .btn-icon,
.c2.primary[aria-disabled=true] .btn-icon {
  color: #8c959f;
}

.c2.primary:focus,
.c2.primary.focus {
  background-color: #2da44e;
  border-color: rgba(208,215,222,0.5);
  box-shadow: 0 0 0 3px rgba(45,164,78,0.4);
}

.c2.primary .btn-icon {
  color: #ffffff;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .meta {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 8px;
}

.c0 .btn-group {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .btn-group .c1 {
  margin-right: 8px;
}

.c0 .btn-group .c1:last-child {
  margin-right: 0;
}

.c3 {
  display: block;
  width: 100%;
  overflow-x: auto;
}

.c3 table {
  width: 100%;
  max-width: 100%;
}

.c3 table .editable div {
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.c3 table .editable svg {
  color: #6e7781;
}

.c3 table {
  border: none;
  height: auto;
  border-collapse: separate;
  border-spacing: 0 8px;
}

.c3 table th,
.c3 table td {
  width: 1%;
  vertical-align: middle !important;
}

.c3 table th {
  border: none;
  border-right: 1px solid rgba(208,215,222,0.5) !important;
}

.c3 table th:last-child,
.c3 table td:last-child {
  border-right: none !important;
}

.c3 table td {
  border: none;
}

.c3 table thead {
  font-weight: 500;
}

.c3 table thead th {
  text-align: center;
}

.c3 table thead th:first-child {
  text-align: left;
}

.c3 table tbody tr {
  background-color: #ffffff;
  box-shadow: 0 3px 6px rgba(140,149,159,0.15),inset 0 1px 0 rgba(255,255,255,0.25);
}

.c3 table tbody {
  --border-radius-cell: 6px;
}

.c3 table tbody tr td:first-child {
  border-top-left-radius: var(--border-radius-cell);
  border-bottom-left-radius: var(--border-radius-cell);
}

.c3 table tbody tr td:last-child {
  border-top-right-radius: var(--border-radius-cell);
  border-bottom-right-radius: var(--border-radius-cell);
}

.c4 {
  border-spacing: 0;
  border: 1px solid #d0d7de;
}

.c4 tr:last-child td {
  border-bottom: 0;
}

.c4 th,
.c4 td {
  margin: 0;
  padding: 0.5rem;
  border-bottom: 1px solid #d0d7de;
  border-right: 1px solid #d0d7de;
}

.c4 th:last-child,
.c4 td:last-child {
  border-right: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  padding: 0 0;
}

.c5 .editable {
  overflow: auto;
}

.c5 .toolbar {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  box-shadow: 0 1px 0 rgba(27,31,36,0.04);
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c6 div[disabled],
.c6 div.disabled,
.c6 div[aria-disabled=true] {
  pointer-events: none;
  color: #8c959f;
  opacity: 65%;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 24px;
  min-width: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 3px;
  padding: 4px;
  color: #222f3e;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c8:hover {
  background-color: #dee0e2;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  max-height: 200px;
  overflow-x: hidden;
  overflow-y: auto;
  padding: 0;
}

@media (min-width:767px) {
  .c3 table tbody td .toolbar {
    visibility: hidden;
    opacity: 0;
    -webkit-transition: visibility 0s,opacity 0.1s linear;
    transition: visibility 0s,opacity 0.1s linear;
  }

  .c3 table tbody td:hover .toolbar {
    visibility: visible;
    opacity: 1;
  }
}

<div
  class="c0"
>
  <div
    class="meta"
  >
    <h2
      class="h2"
    >
      Players
    </h2>
    <div
      class="btn-group"
    >
      <button
        class="c1 c2 primary"
        disabled=""
      >
        Edit
      </button>
    </div>
  </div>
  <div
    class="c3"
  >
    <table
      class="c4 table"
      role="table"
    >
      <thead>
        <tr
          role="row"
        >
          <th
            colspan="1"
            role="columnheader"
          >
            Player
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            Game 1
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            Ade
          </td>
          <td
            role="cell"
          >
            <div
              class="c5"
              data="[object Object]"
              headers="[object Object],[object Object]"
            >
              <div
                class="c6 editable"
              >
                <div
                  class="c7"
                >
                  <div
                    aria-disabled="false"
                    class="c8"
                    data-item="H"
                  >
                    <div
                      class="c9"
                    >
                      H
                    </div>
                  </div>
                  <div
                    aria-disabled="false"
                    class="c8 toolbar"
                    data-class="toolbar"
                    data-item="toolbar"
                  >
                    <div
                      class="c9"
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-plus fa-w-14 "
                        data-icon="plus"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        viewBox="0 0 448 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Display FinesGroup should create valid open modal snapshot 1`] = `
.c2 {
  position: relative;
  display: inline-block;
  padding: 5px 16px;
  font-size: 14px;
  font-weight: 600;
  line-height: 20px;
  white-space: nowrap;
  vertical-align: middle;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid;
  border-radius: 6px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c2:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2:disabled,
.c2.disabled,
.c2[aria-disabled=true] {
  cursor: default;
}

.c2 .dropdown-caret {
  margin-left: 4px;
  opacity: 0.8;
}

.c2 .btn-icon {
  margin-right: 4px;
  color: inherit;
  vertical-align: text-bottom;
}

.c2 .btn-icon:only-child {
  margin-right: 0;
}

.c2 {
  color: #24292f;
  background-color: #f6f8fa;
  border: 1px rgba(208,215,222,0.5) solid;
  box-shadow: 0 1px 0 rgba(27,31,36,0.04),inset 0 1px 0 rgba(255,255,255,0.25);
  -webkit-transition: 0.2s cubic-bezier(0.3,0,0.5,1);
  transition: 0.2s cubic-bezier(0.3,0,0.5,1);
  -webkit-transition-property: color,background-color,border-color;
  transition-property: color,background-color,border-color;
}

.c2:hover,
.c2.hover,
[open] > .c2 {
  background-color: #eaeef2;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
}

.c2:active {
  background-color: #e1e6ec;
  -webkit-transition: none;
  transition: none;
}

.c2.selected,
.c2[aria-selected=true] {
  background-color: #e4e9ee;
  box-shadow: inset 0 1px 0 rgba(208,215,222,0.2);
}

.c2:disabled,
.c2.disabled,
.c2[aria-disabled=true] {
  color: #8c959f;
  background-color: #f6f8fa;
  cursor: default;
}

.c2:disabled .btn-icon,
.c2.disabled .btn-icon,
.c2[aria-disabled=true] .btn-icon {
  color: #8c959f;
}

.c2:focus,
.c2.focus {
  outline: none;
  box-shadow: 0 0 0 3px rgba(9,105,218,0.15);
}

.c2.primary {
  color: #ffffff;
  background-color: #1a7f37;
  border-color: rgba(208,215,222,0.5);
  box-shadow: 0 1px 0 rgba(27,31,36,0.1), inset 0 1px 0 rgba(255,255,255,0.3);
}

.c2.primary:hover,
.c2.primary.hover,
[open] > .c2.primary {
  background-color: #116329;
  border-color: rgba(208,215,222,0.5);
}

.c2.primary:active,
.c2.primary.selected,
.c2.primary[aria-selected=true] {
  background-color: #105a25;
  box-shadow: inset 0 1px 0 rgba(0,45,17,0.2);
}

.c2.primary:disabled,
.c2.primary.disabled,
.c2.primary[aria-disabled=true] {
  color: rgba(255,255,255,0.8);
  background-color: #94d3a2;
  border-color: rgba(208,215,222,0.5);
}

.c2.primary:disabled .btn-icon,
.c2.primary.disabled .btn-icon,
.c2.primary[aria-disabled=true] .btn-icon {
  color: #8c959f;
}

.c2.primary:focus,
.c2.primary.focus {
  background-color: #2da44e;
  border-color: rgba(208,215,222,0.5);
  box-shadow: 0 0 0 3px rgba(45,164,78,0.4);
}

.c2.primary .btn-icon {
  color: #ffffff;
}

.c19 {
  position: relative;
  display: inline-block;
  padding: 5px 16px;
  font-size: 14px;
  font-weight: 600;
  line-height: 20px;
  white-space: nowrap;
  vertical-align: middle;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid;
  border-radius: 6px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c19:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c19:disabled,
.c19.disabled,
.c19[aria-disabled=true] {
  cursor: default;
}

.c19 .dropdown-caret {
  margin-left: 4px;
  opacity: 0.8;
}

.c19 .btn-icon {
  margin-right: 4px;
  color: inherit;
  vertical-align: text-bottom;
}

.c19 .btn-icon:only-child {
  margin-right: 0;
}

.c19 {
  color: #24292f;
  background-color: #f6f8fa;
  border: 1px rgba(208,215,222,0.5) solid;
  box-shadow: 0 1px 0 rgba(27,31,36,0.04),inset 0 1px 0 rgba(255,255,255,0.25);
  -webkit-transition: 0.2s cubic-bezier(0.3,0,0.5,1);
  transition: 0.2s cubic-bezier(0.3,0,0.5,1);
  -webkit-transition-property: color,background-color,border-color;
  transition-property: color,background-color,border-color;
}

.c19:hover,
.c19.hover,
[open] > .c19 {
  background-color: #eaeef2;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
}

.c19:active {
  background-color: #e1e6ec;
  -webkit-transition: none;
  transition: none;
}

.c19.selected,
.c19[aria-selected=true] {
  background-color: #e4e9ee;
  box-shadow: inset 0 1px 0 rgba(208,215,222,0.2);
}

.c19:disabled,
.c19.disabled,
.c19[aria-disabled=true] {
  color: #8c959f;
  background-color: #f6f8fa;
  cursor: default;
}

.c19:disabled .btn-icon,
.c19.disabled .btn-icon,
.c19[aria-disabled=true] .btn-icon {
  color: #8c959f;
}

.c19:focus,
.c19.focus {
  outline: none;
  box-shadow: 0 0 0 3px rgba(9,105,218,0.15);
}

.c19.danger:hover,
.c19.danger.hover,
[open] > .c19.danger {
  color: #ffffff;
  background-color: #a40e26;
  border-color: rgba(208,215,222,0.5);
  box-shadow: 0 1px 0 rgba(27,31,36,0.1), inset 0 1px 0 rgba(255,255,255,0.03);
}

.c19.danger:hover .btn-icon,
.c19.danger.hover .btn-icon,
[open] > .c19.danger .btn-icon {
  color: #ffffff;
}

.c19.danger:active,
.c19.danger.selected,
.c19.danger[aria-selected=true] {
  color: #ffffff;
  background-color: #c2202b;
  border-color: rgba(208,215,222,0.5);
  box-shadow: inset 0 1px 0 rgba(76,0,20,0.2);
}

.c19.danger:disabled,
.c19.danger.disabled,
.c19.danger[aria-disabled=true] {
  color: rgba(207,34,46,0.5);
  background-color: #f6f8fa;
  border-color: rgba(208,215,222,0.5);
  box-shadow: none;
}

.c19.danger:disabled .btn-icon,
.c19.danger.disabled .btn-icon,
.c19.danger[aria-disabled=true] .btn-icon {
  color: #8c959f;
}

.c19.danger:focus,
.c19.danger.focus {
  border-color: rgba(208,215,222,0.5);
  box-shadow: 0 0 0 3px rgba(164,14,38,0.4);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .meta {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 8px;
}

.c0 .btn-group {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .btn-group .c1 {
  margin-right: 8px;
}

.c0 .btn-group .c1:last-child {
  margin-right: 0;
}

.c3 {
  display: block;
  width: 100%;
  overflow-x: auto;
}

.c3 table {
  width: 100%;
  max-width: 100%;
}

.c3 table .editable div {
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.c3 table .editable svg {
  color: #6e7781;
}

.c3 table {
  border: none;
  height: auto;
  border-collapse: separate;
  border-spacing: 0 8px;
}

.c3 table th,
.c3 table td {
  width: 1%;
  vertical-align: middle !important;
}

.c3 table th {
  border: none;
  border-right: 1px solid rgba(208,215,222,0.5) !important;
}

.c3 table th:last-child,
.c3 table td:last-child {
  border-right: none !important;
}

.c3 table td {
  border: none;
}

.c3 table thead {
  font-weight: 500;
}

.c3 table thead th {
  text-align: center;
}

.c3 table thead th:first-child {
  text-align: left;
}

.c3 table tbody tr {
  background-color: #ffffff;
  box-shadow: 0 3px 6px rgba(140,149,159,0.15),inset 0 1px 0 rgba(255,255,255,0.25);
}

.c3 table tbody {
  --border-radius-cell: 6px;
}

.c3 table tbody tr td:first-child {
  border-top-left-radius: var(--border-radius-cell);
  border-bottom-left-radius: var(--border-radius-cell);
}

.c3 table tbody tr td:last-child {
  border-top-right-radius: var(--border-radius-cell);
  border-bottom-right-radius: var(--border-radius-cell);
}

.c4 {
  border-spacing: 0;
  border: 1px solid #d0d7de;
}

.c4 tr:last-child td {
  border-bottom: 0;
}

.c4 th,
.c4 td {
  margin: 0;
  padding: 0.5rem;
  border-bottom: 1px solid #d0d7de;
  border-right: 1px solid #d0d7de;
}

.c4 th:last-child,
.c4 td:last-child {
  border-right: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  padding: 0 0;
}

.c5 .editable {
  overflow: auto;
}

.c5 .toolbar {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  box-shadow: 0 1px 0 rgba(27,31,36,0.04);
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c6 div[disabled],
.c6 div.disabled,
.c6 div[aria-disabled=true] {
  pointer-events: none;
  color: #8c959f;
  opacity: 65%;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 24px;
  min-width: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 3px;
  padding: 4px;
  color: #222f3e;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c8:hover {
  background-color: #dee0e2;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  max-height: 200px;
  overflow-x: hidden;
  overflow-y: auto;
  padding: 0;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-width: 0;
  text-align: left;
  text-transform: none;
  color: #24292f;
}

.c16 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: 650px;
  overflow: auto;
  padding: 16px 16px;
}

.c13 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: none;
  color: #24292f;
  padding: 16px 16px 0 16px;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-top: 1px rgba(208,215,222,0.5) solid;
  padding: 8px 16px;
}

.c17 button {
  margin-left: 8px;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
  background-color: #ffffff;
  border: 1px rgba(208,215,222,0.5) solid;
  border-radius: 6px;
  box-shadow: 0 1px 0 rgba(27,31,36,0.04);
}

.c10 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1100;
}

.c11 {
  position: absolute;
  background-color: rgba(246,248,250,0.6);
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1;
}

@media (min-width:767px) {
  .c3 table tbody td .toolbar {
    visibility: hidden;
    opacity: 0;
    -webkit-transition: visibility 0s,opacity 0.1s linear;
    transition: visibility 0s,opacity 0.1s linear;
  }

  .c3 table tbody td:hover .toolbar {
    visibility: visible;
    opacity: 1;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="meta"
    >
      <h2
        class="h2"
      >
        Players
      </h2>
      <div
        class="btn-group"
      >
        <button
          class="c1 c2 primary"
          disabled=""
        >
          Edit
        </button>
      </div>
    </div>
    <div
      class="c3"
    >
      <table
        class="c4 table"
        role="table"
      >
        <thead>
          <tr
            role="row"
          >
            <th
              colspan="1"
              role="columnheader"
            >
              Player
            </th>
            <th
              colspan="1"
              role="columnheader"
            >
              Game 1
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            role="row"
          >
            <td
              role="cell"
            >
              Ade
            </td>
            <td
              role="cell"
            >
              <div
                class="c5"
                data="[object Object]"
                headers="[object Object],[object Object]"
              >
                <div
                  class="c6 editable"
                >
                  <div
                    class="c7"
                  >
                    <div
                      aria-disabled="false"
                      class="c8"
                      data-item="H"
                    >
                      <div
                        class="c9"
                      >
                        H
                      </div>
                    </div>
                    <div
                      aria-disabled="false"
                      class="c8 toolbar"
                      data-class="toolbar"
                      data-item="toolbar"
                    >
                      <div
                        class="c9"
                      >
                        <svg
                          aria-hidden="true"
                          class="svg-inline--fa fa-plus fa-w-14 "
                          data-icon="plus"
                          data-prefix="fas"
                          focusable="false"
                          role="img"
                          viewBox="0 0 448 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                            fill="currentColor"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    id="modal-root"
  >
    <div
      class="c10"
    >
      <style>
        
          .__react_component_tooltip { z-index: 1103; }
        
      </style>
      <div
        class="c11"
      />
      <div
        class="c12 open selector-select-fines"
        style="align-self: center; margin: 8px auto; min-width: 200px; z-index: 2;"
      >
        <div
          class="c13"
        >
          <h2
            class="h2"
          >
            Select Fines
          </h2>
        </div>
        <div
          class="c14"
        >
          <div
            class="c15"
          >
            <div
              class="c16"
            >
              <div
                class="c6"
              >
                <div
                  class="c7"
                >
                  <div
                    aria-disabled="false"
                    class="c8"
                    data-item="[object Object]"
                    data-tip="Hanging"
                  >
                    <div
                      class="c9"
                    >
                      H
                    </div>
                  </div>
                  <div
                    aria-disabled="false"
                    class="c8"
                    data-item="[object Object]"
                    data-tip="Reverse"
                  >
                    <div
                      class="c9"
                    >
                      R
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c17"
        >
          <div
            class="c18"
          />
          <div
            class="c18"
          >
            <button
              class="c1 c19 danger"
            >
              Close
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
